File: springboot-app/src/main/java/de/ngxa/restaurant/entity/OrderLine.java
package de.ngxa.restaurant.entity;

import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Data
@EqualsAndHashCode(callSuper = true)
@Table(name="NGXA_SHOP_ORDER_LINE")
public class OrderLine extends BaseEntity {
	
	//for internal order
	private String itemName;
	private String itemDesc;
	private String itemNumber;
	
	@OneToMany(fetch= FetchType.EAGER, cascade = CascadeType.ALL)
	@JoinColumn(name="order_line_id", referencedColumnName="id")
	private List<OrderOption> options = new ArrayList<>();
	
	//for external orderlines 
	private String externalItemName;
	private String externalItemDesc;
	private String externalItemNumber;
	private double externalItemPrice = 0.0;
	private int quantity = 0;
	
	@Column(nullable = false, columnDefinition = "boolean default true")
	private boolean isInternal = true;

}

File: manager-app/src/app/dto/order-line.ts
import {BaseDto} from './base-dto';
import {OrderOption} from './order-option';

export class OrderLine extends BaseDto {

  // for internal order
  itemName: string;
  itemDesc: string;
  itemNumber: string;

  options: Array<OrderOption> = [];

  // for external orderlines
  externalItemName: string;
  externalItemDesc: string;
  externalItemNumber: string;
  externalItemPrice = 0.0;
  quantity = 0;

  isInternal = true;
}
